File: course1_C/problem1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

void print_pyramid(int size);

int main(void) 
{
    int size;
    do
    {
        size = get_int("Give height of the pyramid: ");
    }
    /* condition that makes the input correct */
    while (size < 1 || size > 8);

    /* "pseudocode" */
    print_pyramid(size);

    return 0;
}

void print_pyramid(int size)
{
    /* for each line */
    for (int i = 0; i < size; i++)
    {
        /**print spaces before hashes;
         * it refers to the line number as reference
         */
        for (int j = 0; j < size - i - 1; j++) {
            printf(" ");
        }
        
        /* print hashes; again line no as reference */
        for (int j = 0; j < i + 1; j++) {
            printf("#");
        }
        /* print the two spaces between the hashes, fixed */
        printf("  "); 
        
        /* print hashes again; same as above */
        for (int j = 0; j < i + 1; j++) {
            printf("#");
        }
        /* move to the next line, no need for extra spaces */
        printf("\n"); 
    }
}

File: course6_Python/problem6/sentimental-mario-more/mario.py
def main():
    # get pyramid height with correct check - implement do-while loop
    while True:
        try:
            pyramid_height = int(input("Give height of the pyramid: "))
            # accept answer if only:
            if pyramid_height > 0 and pyramid_height < 9:
                break
        except:
            ...

    print_pyramid(pyramid_height)


def print_pyramid(size):
    """print pyramid per the input size, where size is  pyramid height"""
    # for each line
    for i in range(size):
        # initialize string
        line = ""
        # add the necessary spaces before the hashes for the mirroring effect
        for j in range(size - i - 1):
            line += " "
        # print the necessary number of hashes
        for hash in range(i + 1):
            line += "#"
        # add two spaces
        line += "  "
        # and repeat the hashes for the mirroring effect
        for hash in range(i + 1):
            line += "#"
        # and print:
        print(line)


# set up main
if __name__ == "__main__":
    main()
